sdk/core/core-http/test/policies/deserializationPolicyTests.ts

```diff
@@ -808,6 +808,34 @@ describe("deserializationPolicy", function() {
       }
     });
 
+    it(`should throw when the response code is not defined in the operationSpec`, async function() {
+      const serializer = new Serializer(undefined, true);
+
+      const operationSpec: OperationSpec = {
+        httpMethod: "GET",
+        responses: {
+          200: {}
+        },
+        serializer
+      };
+
+      const response: HttpOperationResponse = {
+        request: createRequest(operationSpec),
+        status: 400,
+        headers: new HttpHeaders(),
+        bodyAsText: '{"message": "InternalServerError"}'
+      };
+
+      try {
+        await deserializeResponse(response);
+        assert.fail();
```

Comment:
perhaps having a message here would make the failure recognizable if it were to happen.